osf/metadata/serializers/google_dataset_json_ld.py

```diff
@@ -0,0 +1,65 @@
+import json
+from osf.metadata import gather
+from osf.metadata.serializers import _base
+from osf.metadata.rdfutils import DCT, OSF, FOAF
+from website import settings
+
+class GoogleDatasetJsonLdSerializer(_base.MetadataSerializer):
+    mediatype = 'application/ld+json'
+
+    def filename(self, osfguid: str):
+        return f'{osfguid}-metadata.json-ld'
+
+    def serialize(self, basket: gather.Basket):
+        metadata = {
+            '@context': 'https://schema.org',
+            '@type': 'Dataset',
+            'dateCreated': next(basket[DCT.created]),
+            'dateModified': next(basket[DCT.modified]),
+            'name': next(basket[DCT.title]),
+            'description': next(basket[DCT.description], None),
+            'url': next(basket[DCT.identifier]),
```

Comment:
I want the osf.io URL, how should I filter them? The generator to my knowledge should always yield the osf.io url first.

Reply:
could `next(url for url in basket[DCT.identifier] if url.startswith(OSFIO))`? want to encourage low-knowledge serializers and unordered sets, make changing generator order a safe thing to do

Reply:
could also include as an array, so doi-urls are included as well: `list(basket[DCT.identifier])`

Reply:
I feel like we have readability problems with `next(url for url in basket[DCT.identifier] if url.startswith(OSFIO))` It's not clear to me why I shouldn't write a new gatherer and just go directly with `next(basket[OSF.url])` seems like I shouldn't ever have to yield things and skip over them.

Reply:
i agree that's clunky... a few ways i think we could make it better:
- `[]` instead of `next()`, json-ld likes lists just fine
- multiple lines like a haiku (tho getting into readability matters-of-taste)
- (my favorite) don't skip over things, they were labeled `dcterms:identifier` for a reason (are urls that resolve to the same resource)

so i guess i'd suggest
```suggestion
            'url': list(basket[DCT.identifier]),
```